### PR TITLE
raftstore-v2: fix compact range bugs that causes false positive clean tablet

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Ref #12842 
What's Changed:

```commit-message
- Fix a bug of compact range that causes a dirty tablet being reported as clean.
- Added an additional check to ensure trim's correctness.
- Fix a bug that some tablets are not destroyed and block peer destroy progress.
```

### Related changes

https://github.com/tikv/rocksdb/pull/345
or https://github.com/tikv/rocksdb/pull/346

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -207,6 +207,8 @@ procinfo = { git = "https://github.com/tikv/procinfo-rs", rev = "6599eb9dca74229
 # After the PR to rust-rocksdb is merged, remember to comment this out and run `cargo update -p rocksdb`.
 # [patch.'https://github.com/tikv/rust-rocksdb']
 # rocksdb = { git = "https://github.com/your_github_id/rust-rocksdb", branch = "your_branch" }
+[patch.'https://github.com/tikv/rust-rocksdb']
+rocksdb = { git = "https://github.com/tabokie/rust-rocksdb", branch = "pick-compaction" }
```
Comment (tonyxuqqi) [inline Cargo.toml:211]: make sure this is reverted before it's merged.

**Review by tonyxuqqi — APPROVED** [review]

**Review by SpadeA-Tang — APPROVED** [review]

**Pull request merged**